main.go

```diff
@@ -514,6 +508,44 @@ func main() {
 	}
 }
 
+func registerSelf(ctx context.Context, conf *config.Config, logger *zap.SugaredLogger) func(ctx context.Context, c *config.Config) error {
+	var registerClient = k8s.AdmissionWebhookRegisterClient{
+		Logger: logger.Named("admissionWebhookRegisterClient"),
+	}
+
+	err := registerClient.Register(ctx, conf)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	return registerClient.Unregister
+}
+
+func prepareTLSConfig(ctx context.Context, conf *config.Config, logger *zap.SugaredLogger) *tls.Config {
+	webhookMode := conf.WebhookMode
+
+	tlsConfig := &tls.Config{
+		MinVersion: tls.VersionTLS12,
+	}
+
+	switch {
+	case strings.EqualFold(webhookMode, config.ModeAuto):
+		tlsConfig.Certificates = append([]tls.Certificate(nil), conf.GetOrResolveCertificate())
+	case strings.EqualFold(webhookMode, config.ModeManual):
+		source, err := workloadapi.NewX509Source(ctx)
+		if err != nil {
+			logger.Fatalf("error getting x509 source: %v", err.Error())
+			return nil
+		}
+		tlsConfig.GetCertificate = tlsconfig.GetCertificate(source)
+	default:
+		logger.Fatalf("unknown webhook mode selected: %v", webhookMode)
+		return nil
+	}
```

Comment:
```suggestion
	switch strings.ToUpper(webhookMode) {
	case config.ModeAuto:
		tlsConfig.Certificates = append([]tls.Certificate(nil), conf.GetOrResolveCertificate())
	case config.ModeManual:
		source, err := workloadapi.NewX509Source(ctx)
		if err != nil {
			logger.Fatalf("error getting x509 source: %v", err.Error())
			return nil
		}
		tlsConfig.GetCertificate = tlsconfig.GetCertificate(source)
	default:
		logger.Fatalf("unknown webhook mode selected: %v", webhookMode)
		return nil
	}
```

Reply:
We can probably create a separate function that would parse `webhookMode` and convert it to some enum, similar to how logrus does it in `logrus.ParseLevel`.
Especially if we want to have more than 2 possible values.

Reply:
Thanks, fixed